File: crates/dewasm-backend-perl/units/memory/grow.pl
# requires: memory/size
sub grow {
    my ($self, $delta) = @_;
    my $old = $self->size();
    return 0xFFFFFFFF if $old + $delta > $self->{max_pages};
    $self->{data} .= "\0" x ($delta * 65536);
    return $old;
}

File: crates/dewasm-backend-go/units/memory/grow.go
// requires: memory/size
func (m *Memory) grow(delta uint32) uint32 {
    old := m.size()
    if uint64(old)+uint64(delta) > uint64(m.maxPages) {
        return 0xffffffff
    }
    m.data = append(m.data, make([]byte, uint64(delta)*65536)...)
    return old
}
